<p>When the call to a function doesn't have any side effects, what is the point of making the call if the results are ignored? In such case, either
the function call is useless and should be dropped or the source code doesn't behave as expected.</p>
<p>This rule raises an issue when a builtin function or methods which has no side effects is called and its result is not used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
list([1, 2])  # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
myvar = list([1, 2])
</pre>
<h2>Exceptions</h2>
<p>No issue will be raised when the function or method call is the only statement in a <code>try...except</code> body. This usually indicates that an
exception is expected, and this exception is the side-effect.</p>
<pre>
def tryExcept():
    d = {}
    try:
        d[1]
    except IndexError as e:
        pass

    try:
        int("abc")
    except ValueError as e:
        pass
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://docs.python.org/3/library/functions.html">Python documentation - Built-in Functions</a> </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/9YIRAQ">CERT, EXP12-C.</a> - Do not ignore values returned by functions </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/9gEqAQ">CERT, EXP00-J.</a> - Do not ignore values returned by methods </li>
</ul>

